### PR TITLE
ncm-ntpd: Add support for the 'interface' directive

### PR DESCRIPTION
**Pull request by gombasg**
See the title.

### CODE REVIEW

File: ncm-ntpd/src/main/pan/components/ntpd/schema.pan [hunk 1]
```diff
@@ -96,6 +96,14 @@ type ntpd_system_options = {
     "stats" ? boolean
 };
 
+@documentation{
+    Defines a single rule for the "interface" statement. See the ntp_misc manpage.
+}
+type ntpd_interface_options = {
+    "action" : string with match(SELF, "^(listen|ignore|drop)$")
```
Comment (stdweird) [inline ncm-ntpd/src/main/pan/components/ntpd/schema.pan]: use `choice`

File: ncm-ntpd/src/main/perl/ntpd.pm [hunk 2]
```diff
@@ -264,6 +264,12 @@ sub write_ntpd_config
         print $fh "restrict " . $$client[0] . " mask " . $$client[1] . " nomodify notrap\n";
     }
 
+    if ($cfg->{interface}) {
+        foreach my $rule (@{$cfg->{interface}}) {
+            print $fh "interface " . $rule->{action} . " " . $rule->{match} . "\n";
```
Comment (stdweird) [inline ncm-ntpd/src/main/perl/ntpd.pm]: put the whle thing in one `"..."`, see eg lines below

**Comment by stdweird** [comment]
jenkins test this please

**Comment by stdweird** [comment]
jenkins test this please

File: ncm-ntpd/src/main/pan/components/ntpd/schema.pan (hunk 1 above)
Comment (ned21) [inline ncm-ntpd/src/main/pan/components/ntpd/schema.pan:103]:
```suggestion
    "action" : string with choice("listen", "ignore", "drop")
```

File: ncm-ntpd/src/main/perl/ntpd.pm (hunk 2 above)
Comment (ned21) [reply ncm-ntpd/src/main/perl/ntpd.pm:269]:
```suggestion
            print $fh "interface $rule->{action} $rule->{match}\n";
```

**Comment by ned21** [comment]
Trying to edit this via the web interface was a disaster, and I can't see any easy way to drop the bad commit. Have replaced it.

**Pull request closed without merging**